### PR TITLE
Migrating snap auto delete test

### PR DESCRIPTION
**Pull request by srijan-sivakumar**
Test being migrated,
    [test_auto_delete](https://github.com/gluster/glusto-tests/blob/master/tests/functional/snapshot/test_auto_delete.py)

Updates: #292

### CODE REVIEW

File: tests/example/sample_component/test_snap_ops.py [hunk 1]
```diff
@@ -2,6 +2,7 @@
 This file contains a test-case which tests
 the snapshot ops.
 """
+# disruptive;rep
 # disruptive;rep,dist,dist-rep,arb,dist-arb,disp,dist-disp
```
Comment (nik-redhat) [inline tests/example/sample_component/test_snap_ops.py:6]: 2 comments?

File: tests/example/sample_component/test_snap_ops.py [hunk 2]
```diff
@@ -26,6 +27,11 @@ def run_test(self, redant):
                            self.server_list[0])
         redant.snap_create(self.vol_name, self.snap_name2,
                            self.server_list[0])
+
+        val = redant.get_snap_config(self.server_list[0], self.vol_name)
+        print(val)
```
Comment (nik-redhat) [inline tests/example/sample_component/test_snap_ops.py:32]: Ig it's not needed

File: tests/functional/snapshot/test_auto_delete.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+"""
```
Comment (nik-redhat) [inline tests/functional/snapshot/test_auto_delete.py:17]: nit: description

File: tests/functional/snapshot/test_auto_delete.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        # Set snap-max-soft-limit snapshot config option
+        max_soft_limit = {'snap-max-soft-limit': '80'}
+        redant.set_snap_config(max_soft_limit, self.server_list[0])
+
+        # Validate snap-max-soft-limit snapshot config option
+        config_v = redant.get_snap_config(self.server_list[0],
+                                          self.vol_name)
+        if config_v['volumeConfig'][0]['softLimit'] != '8':
+            raise Exception("Expected softlimit of 8 but got "
+                            f" {config_v['volumeConfig'][0]['softLimit']} for"
```
Comment (nik-redhat) [inline tests/functional/snapshot/test_auto_delete.py:87]: nit: extra space

**Review by nik-redhat — CHANGES_REQUESTED** [review]

**Comment by nik-redhat** [comment]
Can you please attach the output as well?

**Comment by srijan-sivakumar** [comment]
```console
Results for snapshot
Test nature : disruptive
TestName : test_auto_delete
+-------------+-------------+-----------------------+-------------+
| Volume Type | Test Result | Time Taken (hh:mm:ss) | Skip Reason |
+-------------+-------------+-----------------------+-------------+
|     rep     |     PASS    |        00:05:22       |      NA     |
|   dist-rep  |     PASS    |        00:06:05       |      NA     |
|     disp    |     PASS    |        00:03:46       |      NA     |
|  dist-disp  |     PASS    |        00:05:52       |      NA     |
|     dist    |     PASS    |        00:03:52       |      NA     |
+-------------+-------------+-----------------------+-------------+
+--------------------------------------+-------+
|               Category               | Count |
+--------------------------------------+-------+
|           Disruptive Tests           |   1   |
|         Non Disruptive Tests         |   0   |
|       Disruptive Tests Skipped       |   0   |
|     Non Disruptive Tests Skipped     |   0   |
|              Tests Ran               |   1   |
|            Tests Skipped             |   0   |
|             Total Tests              |   1   |
|   Disruptive Tests Pass Percentage   | 100.0 |
| Non Disruptive Tests Pass Percentage |   0   |
|        Total Pass Percentage         | 100.0 |
+--------------------------------------+-------+
Total Time Taken : 00 hours 25 minutes 10 seconds
✔ Tearing down successful.
```

**Review by nik-redhat — APPROVED** [review]

**Pull request merged**